tornado/gen.py

```diff
@@ -698,108 +683,81 @@ class Runner(object):
     """
 
     def __init__(
-        self,
-        gen: "Generator[_Yieldable, Any, _T]",
-        result_future: "Future[_T]",
-        first_yielded: _Yieldable,
+        self, gen: "Generator[_Yieldable, Any, _T]", result_future: "Future[_T]"
     ) -> None:
         self.gen = gen
         self.result_future = result_future
         self.future = _null_future  # type: Union[None, Future]
         self.running = False
         self.finished = False
         self.io_loop = IOLoop.current()
-        if self.handle_yield(first_yielded):
-            gen = result_future = first_yielded = None  # type: ignore
-            self.run()
+        self.task = asyncio.ensure_future(self.run())
+
+    async def run(self) -> None:
+        "Runs the generator to completion in the context of a task"
+        while True:
+            future = self.future
+            if future is None:
+                raise Exception("No pending future")
+            if not self.future.done():  # type: ignore
+                _step = asyncio.Event()
```

Comment:
Style note: Tornado uses a leading underscore on "private" class/instance attributes, but local variables should just have plain names like `step`.